frappe/utils/background_jobs.py

```diff
@@ -276,12 +276,16 @@ def execute_job(site, method, event, job_name, kwargs, user=None, is_async=True,
 		return retval
 
 	finally:
-		for after_job_task in frappe.get_hooks("after_job"):
-			frappe.call(after_job_task, method=method_name, kwargs=kwargs, result=retval)
-		frappe.local.job.after_job.run()
-
-		if is_async:
-			frappe.destroy()
+		if retry == 0:
+			if not hasattr(frappe.local, "site"):
+				frappe.init(site)
+				frappe.connect()
+			for after_job_task in frappe.get_hooks("after_job"):
+				frappe.call(after_job_task, method=method_name, kwargs=kwargs, result=retval)
```

Comment:
Do you know any problematic ones that might fail? I am not really seeing a problem with executing before-after for all retries. They are usually doing some time accounting. 

AFAIK, all of they seem to be idempotent wrt one execution and can be restarted for next execution. 

It should be almost like re-enqueue of entire job IMO.

Reply:
Cool, will remove that check then. Wasn't sure of any specific ones.